framework/Optimizers/SPSA.py

```diff
@@ -332,16 +360,25 @@ def localGenerateInput(self,model,oldInput):
         ck = self._computeGainSequenceCk(self.paramDict,self.counter['varsUpdate'][traj]+1)
         varK = dict((var,self.counter['recentOptHist'][traj][0][var]) for var in self.getOptVars(traj))
         #check the submission queue is empty; otherwise something went wrong # TODO this is a sanity check, might be removed for efficiency
-        #TODO this same check is in GradientBasedOptimizer.queueUpOptPointRuns, they might benefit from abstracting
-        if len(self.submissionQueue[traj]) > 0:
-          self.raiseAnError(RuntimeError,'Preparing to add grad evals to submission queue for trajectory "{}" but it is not empty: "{}"'.format(traj,self.submissionQueue[traj]))
-        for i in self.perturbationIndices: #perturbation points are odd, not even
+        assert(len(self.submissionQueue[traj])==0)
```

Comment:
It is not possible, from the user's side; thus, an assertion, since obviously a future developer could mistakenly interfere and not realize what was destroyed without significant searching.  This assertion helps such a future developer not make that mistake.